frontend/src/lib/v2/StaticFlow.ts

```diff
@@ -271,19 +293,26 @@ function addArtifactToTaskEdges(
     for (let artifactKey in artifacts) {
       const artifactSpec = artifacts[artifactKey];
       const taskOutputArtifact = artifactSpec.taskOutputArtifact;
-      if (!taskOutputArtifact) {
-        continue;
+      const componentInputArtifact = artifactSpec.componentInputArtifact;
+      if (taskOutputArtifact) {
+        const outputArtifactKey = taskOutputArtifact.outputArtifactKey;
+        const producerTask = taskOutputArtifact.producerTask;
+        const edge: Edge = {
+          id: getArtifactToTaskEdgeKey(outputArtifactKey, inputTaskKey),
+          source: getArtifactNodeKey(producerTask, outputArtifactKey),
+          target: getTaskNodeKey(inputTaskKey),
+          arrowHeadType: ArrowHeadType.ArrowClosed,
+        };
+        flowGraph.push(edge);
+      } else if (componentInputArtifact) {
+        const edge: Edge = {
+          id: getArtifactToTaskEdgeKey(componentInputArtifact, inputTaskKey),
+          source: getArtifactNodeKey('', componentInputArtifact),
+          target: getTaskNodeKey(inputTaskKey),
+          arrowHeadType: ArrowHeadType.ArrowClosed,
+        };
+        flowGraph.push(edge);
       }
```

Comment:
Are these two conditions also handle the `input artifact -> subDAG` and `subDAG -> output artifact` ? 

![Screen Shot 2022-10-10 at 10 06 40 AM](https://user-images.githubusercontent.com/56132941/194919167-a93fb533-b976-41ff-81b5-b78f5cc1f4b2.png)

Reply:
`subDAG -> output artifact` is able to be handled already before this PR. `input artifact -> subDAG` support is added in this PR.